src/revocation.rs

```diff
@@ -8,9 +8,99 @@ use crate::error::{Error, Result};
 use crate::secure_mount;
 
 use std::path::Path;
+use std::process::Command;
 
 use serde_json::Value;
 
+/// Runs a python script (used for revocation actions)
+pub(crate) fn run_python(dir: &Path, script: &str) -> Result<()> {
+    let status =
+        // TODO: Should we assume that these script names are all correctly 
+        // formatted and end with .py? Or should we allow some wiggle room and
+        // append .py if it isn't there already?
+        match Command::new("python").current_dir(dir).arg(script).status() {
+            Ok(status) => status,
+            Err(e) => {
+                let msg = format!("ERROR: failed to run revocation action: {:?}", script);
+                // TODO: Unless errors are already logged automatically?
+                error!("{}", msg);
+                return Err(Error::Other(msg));
+            }
+        };
+
+    if !status.success() {
+        let msg = format!(
+            "ERROR: revocation action {:?} returned with status {:?}",
+            script, status
+        );
+        error!("{}", msg);
```

Comment:
Is there any output to `stdout` or `stderr` we may need to capture?

Reply:
Also a great idea, I added this. Thanks!